python/gaas_server/gaas_handler.py

```diff
@@ -477,76 +720,94 @@ def __add_graph(self, G):
         self.__next_graph_id += 1
         return gid
 
-    def __get_dataframe_from_user_props(self, dataframe, columns=None):
+    def __create_graph(self):
         """
+        Instantiate a graph object using a type appropriate for the handler (
+        either SG or MG)
         """
-        internal_columns=[PropertyGraph.vertex_col_name,
-                          PropertyGraph.src_col_name,
-                          PropertyGraph.dst_col_name,
-                          PropertyGraph.type_col_name,
-                          PropertyGraph.edge_id_col_name,
-                          PropertyGraph.vertex_id_col_name,
-                          PropertyGraph.weight_col_name]
+        return MGPropertyGraph() if self.is_mg else PropertyGraph()
 
-        if columns is None or len(columns) == 0:
-            all_user_columns = list(dataframe.columns)
-            # Create a list of user-visible columns by removing the internals while
-            # preserving order
-            for col_name in internal_columns:
-                if col_name in all_user_columns:
-                    all_user_columns.remove(col_name)
-        else:
-            all_user_columns = columns
-        
-        print(all_user_columns)
+    # FIXME: consider adding this to PropertyGraph
+    def __remove_internal_columns(self, pg_column_names):
+        """
+        Removes all column names from pg_column_names that are "internal" (ie.
+        used for PropertyGraph bookkeeping purposes only)
+        """
+        internal_column_names=[PropertyGraph.vertex_col_name,
+                               PropertyGraph.src_col_name,
+                               PropertyGraph.dst_col_name,
+                               PropertyGraph.type_col_name,
+                               PropertyGraph.edge_id_col_name,
+                               PropertyGraph.vertex_id_col_name,
+                               PropertyGraph.weight_col_name]
 
-        # This should NOT be a copy of the dataframe data
-        try:
-            return dataframe[all_user_columns]
-        except KeyError as ke:
-            raise GaasError(f'KeyError({ke.args[-1]})')
+        # Create a list of user-visible columns by removing the internals while
+        # preserving order
+        user_visible_column_names = list(pg_column_names)
+        for internal_column_name in internal_column_names:
+            if internal_column_name in user_visible_column_names:
+                user_visible_column_names.remove(internal_column_name)
+
+        return user_visible_column_names
 
-    def __get_dataframe_rows_as_numpy_bytes(self,
-                                            dataframe,
-                                            index_or_indices,
-                                            null_replacement_value):
+    # FIXME: consider adding this to PropertyGraph
+    def __get_edge_IDs_from_graph_edge_data(self,
```

Comment:
might want to drop another FIXME here for handling the case where the user wants to filter by edge type

Reply:
Done, and added issue: https://github.com/rapidsai/cugraph/issues/2655